.github/workflows/release.yaml

```diff
@@ -0,0 +1,49 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  release:
+    permissions:
+      id-token: write
+      contents: write #  to create release (changesets/action)
+      issues: write # to post issue comments (changesets/action)
+      pull-requests: write #  to create pull request (changesets/action)
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # This makes Actions fetch all Git history so that Changesets can generate changelogs with the correct commits
+          fetch-depth: 0
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+        env:
+          SKIP_YARN_COREPACK_CHECK: '1'
+
+      - name: Install NPM
+        run: |
+          corepack enable
+          corepack prepare
+      - name: Install dependencies
+        run: npm i
+
+      - name: Set auth token
+        run: |
+          echo npmAuthToken: "${{ secrets.NPM_TOKEN }}" >> ./.yarnrc.yml
```

Comment:
This will need to be in a different format, since this repo isn't using yarn. So the `.yarnrc.yml` won't get used. 

I believe with NPM you can just omit this whole step and it should be fine, since you have the NPM_TOKEN set from the secrets in the next step.

Reply:
Correction: this line should look like:
```
echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > .npmrc
```

Reply:
Oh yes good catch 👍